Editors/README.md

```diff
@@ -22,56 +22,32 @@ The [Swift for Visual Studio Code extension](https://marketplace.visualstudio.co
 
 ## Sublime Text
 
-Before using SourceKit-LSP with Sublime Text, you will need to install the LSP package from Package Control. To configure SourceKit-LSP, open the LSP package's settings. The following snippet should be enough to get started with Swift.
+Before using SourceKit-LSP with Sublime Text, you will need to install the LSP package from Package Control. And you're ready to go if you have `xcrun` in the `$PATH`.
 
-You will need the path to the `sourcekit-lsp` executable for the "command" section.
+To configure SourceKit-LSP additionally, open the SourceKit-LSP package's settings by typing in command palette `Preferences: LSP-SourceKit Settings`. The following snippet should be enough to get started with Swift and Objective-C/C++ and the custom path to `sourcekit-lsp` executable.
 
 ```json
 {
-  "clients":
-  {
-    "SourceKit-LSP":
-    {
-      "enabled": true,
-      "command": [
+    "command": [
         "<sourcekit-lsp command>"
```

Comment:
I think we can just use `sourcekit-lsp` here, which should work if you have `sourcekit-lsp` in your PATH, which you should have if you have Xcode installed. We do the same for the `vim` instructions below.

Reply:
I've updated, but isn't this man should be cross-platform? I mean the default client side command is `xcrun sourcekit-lsp` that should just work in one have Xcode installed, therefore this setting is redundant at all. But I'm not sure about the picture on Linux and Windows.

Reply:
My understanding is that if you install Xcode, you get `/usr/bin/sourcekit-lsp` (which is in your PATH), which is essentially a trampoline for `xcrun sourcekit-lsp`. So, if we use `sourcekit-lsp` as the command, it uses that on macOS. On Linux, we expect the toolchain to be in your path as well (which it usually is if you install it to `/usr/bin`, so then you also have `sourcekit-lsp` in your path). Given that this should work for most users, I think it’s good to require one less customization step from users.

Reply:
I have nothing to complain here. I'm fine with either way. Just mentioning, that this block could be skipped completely if the minimalism matters.

Reply:
I think the right course of action would be to:
- Change _LSP-SourceKit_ to run `sourcekit-lsp` instead of `xcrun sourcekit-lsp`. `/usr/bin/sourcekit-lsp` is just a shim around `xcrun sourcekit-lsp`.
- Then we don’t even need to include the `command` here.

Reply:
What about Windows users? it's obviously would be a different path there.

Reply:
My thinking here was that Windows users will need to update the path in both versions, so I thought it’s preferable if it at least works out-of-the-box for those on macOS and Linux.

That being said, `sourcekit-lsp.exe` should be in `PATH` on Windows as well if a Swift toolchain is installed. Is there a way in ST to make the executable name dependent on the OS? This is about adding the `.exe` suffix or not.

Reply:
> > Is there a way in ST to make the executable name dependent on the OS?
> 
> AFAIK, no.

Yes, it is. The plugin code of LSP-SourceKit can dynamically modify command based on OS.

Reply:
> Yes, it is. The plugin code of LSP-SourceKit can dynamically modify command based on OS.
But I think that we don't have to care about that for now. LSP-SourceKit is anyway only available for Mac and Linux currently.

I think it would be good to consider adding support for LSP-SourceKit on Windows as well but that that’s a separate topic. 

> @ahoppen What's about this PR? What do we do now? Do we skipping the whole ST part or replacing it with a placeholder text or leaving it as it is just for information purpose?

I think the key pieces of information we should include in the REAME are:
- To use SourceKit-LSP in Sublime, install the LSP-SourceKit package
- Install Swift-Next for syntactic highlighting
- Note that clang-family languages can be enabled in SourceKit-LSP as well, as discussed here https://github.com/apple/sourcekit-lsp/pull/1184#discussion_r1576129038
> Add instructions in its readme that if the user wants to enable it, he/she has to run "LSP: Enable language server in project" from the command palette.
So users would explicitly opt-in to it per-project.

And yes, I think then we don’t need the snippet at all anymore, which is a great simplification in my opinion.

Reply:
> Note that clang-family languages can be enabled in SourceKit-LSP as well

They are enabled by default as rchl suggested here https://github.com/apple/sourcekit-lsp/pull/1184#discussion_r1576129038 so I'd rather skip this part completely.

So to sum things up a bit:

LSP-SourceKit Sublime text plugin's default setup https://github.com/sublimelsp/LSP-SourceKit/pull/7:
- command `sourcekit-lsp`, (plain, not `/usr/bin/sourcekit-lsp`)
- ``enabled`: false` this server disabled globally in ST, so if user installed it he/she should to toggle it on in advance per command palette or by overriding global LSP-SourceKit settings.
- `scope: c | c++ | objc | objc++ | swift` which means that it's handling broader cland-family language scope (therefore it's global disabled by default).

So considering all of this I'll update this readme with the mention of the LSP-SourceKit itself and a Swift-Next syntax to install, and trowing off all the code snippet.

Reply:
Oh, my understanding was that we enabled it by default for Swift but disabled it by default for all the clang languages. But I assume that’s not possible?

Reply:
Nah, I'd rather leave this to answer to @rchl he definitely better in this 😀

Reply:
What I meant before is what @yaroslavyaroslav said and did.

To have it enabled by default for Swift and for the user to be able to opt-in into other file types, could be done but it would be less user friendly. The user would have to go into LSP-SourceKit settings and override the `selector` manually (we could leave commented-out, alternative selector to make it easier). So essentially what this guide said initially. But if the user does that manually, then he/she will still have the same problem that it might trigger in non-swift projects. Unless user overrides `selector` in project-specific settings (which is also possible) but again, less user friendly than just using Command Palette.

So not sure what's better. I would have to leave the decision to you two. I don't even code in Swift.

Reply:
So again, let's sum things up:

1. The scope could be declared only once in a settings config, so it could be either all or nothing.
2. If we declare the full scope of the server, we should disable it by default, due to potential conflicts with `clang` one.
3. We can declare some limited scope as default (for instance objc/objc++/swift), but then it would take more effort from a user to expand it with c/c++ on his side rather than just send a single command in command palette. The other downside of this decision, is that all objc header files identifies as `c` scope in ST by default, so it's likely would be a cause of an inconsistency issue in a UX with the sourcekit-lsp.

So after all, I believe that this is the best option that we got, to make it less error prone on one hand and to try to keep UX at its best on the other.

Reply:
@ahoppen FYI: the PR https://github.com/sublimelsp/LSP-SourceKit/pull/7 is merged, so it's safe to merge this one now as well.